File: src/Frontend/src/models/Activity.js
export class Activity {
  constructor(data = {}) {
    this.id = data.id || '';
    this.name = data.name || '';
    this.date = data.date ? new Date(data.date) : null;
    this.time = data.time || '';
    this.location = data.location || '';
    this.description = data.description || '';
    this.estimatedCost = data.estimatedCost || 0;
    this.status = data.status || 'planirano'; // planirano, rezervisano, završeno, otkazano
    this.travelPlanId = data.travelPlanId || '';
  }
}

File: src/Services/ActivityService/Models/Activity.cs
using System;

namespace ActivityService.Models
{
    public class Activity
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string Time { get; set; } = string.Empty; // HH:mm format
        public string Location { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal EstimatedCost { get; set; }
        public string Status { get; set; } = "planirano"; // planirano, rezervisano, završeno, otkazano
        public Guid TravelPlanId { get; set; }
    }
}
